lib/libzpool/kernel.c

```diff
@@ -813,11 +819,37 @@ __dprintf(const char *file, const char *func, int line, const char *fmt, ...)
 			(void) printf("%llu ", gethrtime());
 		if (dprintf_find_string("long"))
 			(void) printf("%s, line %d: ", newfile, line);
-		(void) printf("%s: ", func);
+		(void) printf("dprintf: %s: ", func);
 		va_start(adx, fmt);
 		(void) vprintf(fmt, adx);
 		va_end(adx);
 		funlockfile(stdout);
+	} else {
+		/* zfs_dbgmsg is logged for dumping later */
+		size_t size;
+		char *buf, *nl;
+		int i;
+
+		size = 1024;
+		buf = umem_alloc(size, UMEM_NOFAIL);
+		i = snprintf(buf, size, "%s:%d:%s(): ", newfile, line, func);
+
+		if (i < size) {
+			va_start(adx, fmt);
+			(void) vsnprintf(buf + i, size - i, fmt, adx);
+			va_end(adx);
+		}
+
+		/*
+		 * Get rid of trailing newline.
+		 */
+		nl = strrchr(buf, '\n');
+		if (nl != NULL)
+			*nl = '\0';
```

Comment:
This doesn't just get rid of trailing newlines, it gets rid of the last newline, which may have more text after it.

Reply:
See comment above.